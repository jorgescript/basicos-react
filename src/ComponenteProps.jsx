import React from "react";
import PropTypes from "prop-types";

export const ComponenteProps = ({ valor }) => (
  <>
    <p data-testid="test-title">Recibi este valor por Props: {valor}</p>
  </>
);

ComponenteProps.propTypes = {
  valor: PropTypes.string.isRequired,
};

ComponenteProps.defaultProps = {
  valor: "Valor por default",
};
