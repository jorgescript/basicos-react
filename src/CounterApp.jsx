import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [state, setState] = useState(value);
  const handleAdd = () => setState(state + 1);
  const handleSubstract = () => setState(state - 1);
  const handleReset = () => setState(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{state}</h2>
      <button onClick={handleAdd}> +1 </button>
      <button onClick={handleSubstract}> -1 </button>
      <button onClick={handleReset}> Reset </button>
    </>
  );
};
CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 0,
};
