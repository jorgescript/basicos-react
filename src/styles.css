html,
body {
  padding: 70px;
  color: #1d1d1d;
  font-size: 1.3rem;
  background-color: #fafafa;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
    Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
}
