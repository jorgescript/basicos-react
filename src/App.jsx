import { ComponenteProps } from "./ComponenteProps";
import { CounterApp } from "./CounterApp";

export const App = () => {
  const contenido = {
    titulo: "Hola mundo",
    parrafo: "Bienvenido a Reactjs",
  };
  return (
    <>
      <h1>{contenido.titulo}</h1>
      <p>{contenido.parrafo}</p>
      <ComponenteProps valor="Jorge" />
      <CounterApp value={10} />
    </>
  );
};
