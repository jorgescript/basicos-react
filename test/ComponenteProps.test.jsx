import { render, screen } from "@testing-library/react";
import { ComponenteProps } from "../src/ComponenteProps";

describe("Pruebas en el componente ComponenteProps", () => {
  const valor = "Jorge";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<ComponenteProps valor={valor} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el valor en un <p>", () => {
    render(<ComponenteProps valor={valor} />);
    expect(screen.getByTestId("test-title").innerHTML).toBe(
      `Recibi este valor por Props: ${valor}`
    );
  });
});
