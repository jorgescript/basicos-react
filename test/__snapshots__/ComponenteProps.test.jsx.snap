// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ComponenteProps Debe hacer match con el snapshot 1`] = `
<div>
  <p
    data-testid="test-title"
  >
    Recibi este valor por Props: 
    Jorge
  </p>
</div>
`;
